Sources/Sentry/SentryProfiler.mm

```diff
@@ -79,108 +79,6 @@
     return [symbolNSStr substringWithRange:[match rangeAtIndex:1]];
 }
 
-std::mutex _gDataStructureLock;
-
-void
-processBacktrace(const Backtrace &backtrace,
-    NSMutableDictionary<NSString *, NSMutableDictionary *> *threadMetadata,
-    NSMutableDictionary<NSString *, NSDictionary *> *queueMetadata,
-    NSMutableArray<SentrySample *> *samples, NSMutableArray<NSArray<NSNumber *> *> *stacks,
-    NSMutableArray<NSDictionary<NSString *, id> *> *frames,
-    NSMutableDictionary<NSString *, NSNumber *> *frameIndexLookup,
-    NSMutableDictionary<NSString *, NSNumber *> *stackIndexLookup)
-{
-    const auto stack = [NSMutableArray<NSNumber *> array];
-    const auto framesToAdd = [NSMutableArray<NSDictionary<NSString *, id> *> array];
-    for (std::vector<uintptr_t>::size_type backtraceAddressIdx = 0;
-         backtraceAddressIdx < backtrace.addresses.size(); backtraceAddressIdx++) {
-        const auto instructionAddress
-            = sentry_formatHexAddressUInt64(backtrace.addresses[backtraceAddressIdx]);
-
-        const auto frameIndex = frameIndexLookup[instructionAddress];
-        if (frameIndex == nil) {
-            const auto frame = [NSMutableDictionary<NSString *, id> dictionary];
-            frame[@"instruction_addr"] = instructionAddress;
-#    if defined(DEBUG)
-            const auto symbols
-                = backtrace_symbols(reinterpret_cast<void *const *>(backtrace.addresses.data()),
-                    static_cast<int>(backtrace.addresses.size()));
-            frame[@"function"] = parseBacktraceSymbolsFunctionName(symbols[backtraceAddressIdx]);
-#    endif
-            [stack addObject:@(frames.count)];
-            frameIndexLookup[instructionAddress] = @(frames.count);
```

Comment:
In the old implementation the logic was changed to use the `framesToAdd` array as a temporary container before applying the append all at once in the critical section, but that means that `frames.count` will now be incorrect here.